src/libstd/thread/mod.rs

```diff
@@ -112,6 +112,25 @@
 //! will want to make use of some form of **interior mutability** through the
 //! [`Cell`] or [`RefCell`] types.
 //!
+//! ## Naming threads
+//!
+//! Threads are able to have associated names for identification purposes. For example, the thread
+//! name is used in panic messages. By default, spawned threads are unnamed. To specify a name for
+//! a thread, build the thread with [`Builder`] and pass the desired thread name to
+//! [`Builder::name`]. To retrieve the thread name from within the thread, use [`Thread::name`].
```

Comment:
this seem fine? https://github.com/rust-lang/rust/pull/43848/commits/10bd80d79b006b1a81a66e1a10e333eaa38b068c